In [15]:
import os
import tarfile

In [3]:
path_ds = '/nobackup/users/schreurs/project_GAN/dataset'

In [4]:
files = [f for f in os.listdir(path_ds) if os.path.isfile(os.path.join(path_ds, f))]
print("Number of days in dataset = ", len(files))

Number of days in dataset =  772


In [5]:
files[0]

'RAD25_OPER_R___TARRRE__L2__20190826T080500_20190827T080000_0001.tar'

# Unpack the tar files
The download dataset consists of 772 days each packed in a tar file. With the code below the tar files are unpacked into folders with the same name

In [14]:
# Testing path
tar_path = path_ds + '/' + files[0]
print("tar path: ", tar_path)
unpack_folder = tar_path.replace('.tar','')
print("unpack in folder: ", unpack_folder)

tar path:  /nobackup/users/schreurs/project_GAN/dataset/RAD25_OPER_R___TARRRE__L2__20190826T080500_20190827T080000_0001.tar
unpack in folder:  /nobackup/users/schreurs/project_GAN/dataset/RAD25_OPER_R___TARRRE__L2__20190826T080500_20190827T080000_0001


In [ ]:
for file in files:
    tar_path = path_ds + '/' + file
    unpack_folder = tar_path.replace('.tar', '')
    
    tar_file = tarfile.open(tar_path)
    tar_file.extractall(unpack_folder) 
    tar_file.close()